### DISCUSSION TITLE
Pagination with getStaticPaths/getStaticProps as query ?page=2

### DISCUSSION THREAD

**Discussion by marcin-piechaczek** [vercel/next.js#21058] [Help] [answered]
Hey guys, 

Is it possible to use getStaticProps/getStaticPaths with pagination query parameters? I need something like this: 
/[...slug]?page=2 to be a static site.

Is it even possible at the current stage?

Thanks for help!

**Comment by Chmarusso** [comment, marked as answer]
@marcin-piechaczek 

> Is it possible to use getStaticProps/getStaticPaths with pagination query parameters? I need something like this:
/[...slug]?page=2 to be a static site.

It is not possible to pass query strings in getStaticPaths that way. You need to pass the `:page` as `/my-results/1`or handle this query string completely client side.

**Reply by frempongandrews** [reply]
can you be more specific?
